doschema/jsvalidation.py

```diff
@@ -0,0 +1,229 @@
+            if self.fields_types_dict[curr_field].field_type != field_value:
+                err_msg = "Type mismatch in schemas {0} and {1}."
+                old_schema = self.fields_types_dict[curr_field].schema_index
+                raise JSONSchemaCompatibilityException(
+                    old_schema,
+                    self.schema_name,
+                    err_msg.format(old_schema, self.schema_name)
+                )
+        else:
+            self.fields_types_dict[curr_field] = FieldToAdd(
+                schema_index=self.schema_name,
+                field_tuple=curr_field,
+                field_type=field_value
+            )
+
+    def check_special(self, curr_schema, curr_field):
+        """Check schema for allOf, anyOf and oneOf.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        keys = self.COLLECTION_KEYS.intersection(curr_schema)
+        for key in keys:
+            self.collect_special(
+                curr_schema[key],
+                curr_field
+            )
+
+    def collect_special(self, JSON_objects_list, curr_field):
+        """Operate a field with name 'allOf', 'anyOf' or 'oneOf'.
+
+        :param JSON_objects_list: List of JSON objects.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        for elem in JSON_objects_list:
+            self.traverse(elem, curr_field)
+
+    def traverse_dependencies(self, curr_schema, curr_field):
+        """Traverse and check dependencies in schema.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        """
+        properties = curr_schema['properties']
+        dependencies = curr_schema['dependencies']
+
+        while '$ref' in dependencies:
+            path = dependencies.get('$ref')
+            dependencies = self.resolver.resolve(path)[1]
+
+        self.traverse(dependencies, curr_field)
+
+        for field_name, field_value in dependencies.items():
+            if field_name not in properties:
+                continue
+
+            err_msg = "Dependencies error in schema {0}."
+
+            if isinstance(field_value, dict):
+                for name, value in field_value['properties'].items():
+                    if name not in properties:
+                        raise JSONSchemaDependenciesException(
+                            self.schema_name,
+                            err_msg.format(self.schema_name)
+                        )
+                    elif value['type'] != properties[name]['type']:
+                        raise JSONSchemaDependenciesException(
+                            self.schema_name,
+                            err_msg.format(self.schema_name)
+                        )
+            elif type(field_value) is list:
+                for field in field_value:
+                    if field not in properties:
```

Comment:
IMHO it would be better to just test if the field has a type in `fields_types_dict`. It could have been defined somewhere else than in the sibling `properties` dict.